File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.java
class Solution {
    public int[] solution(String s) {
       /*
        int count_step = 0;
        int count_zero = 0;

        while (!s.equals("1")){
            count_zero += s.length() - s.replace("1","").length();
            count_step++;
            s = Integer.toBinaryString(Integer.parseInt(s,2));
        }
        int[] answer = {count_step, count_zero};
        return answer;*/
        int count_step = 0;
        int count_zero = 0;

        while(true){
            if(s.equals("1")) break;
            for(int i = 0; i < s.length(); i++){
                if(s.charAt(i) == "0".charAt(0)) count_zero++;
            }
            s = s.replace("0","");
            s = Integer.toBinaryString(s.length());
            count_step += 1;
        }
        int[] answer = {count_step, count_zero};
        return answer;
    }
}

File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.py
def solution(s):
    count = [0,0]
    while s != "1":
        count[1] += s.count("0")
        s = s.replace("0","")
        count[0] += 1
        s = str(bin(len(s))[2:])
    return count

        
# def solution(s):
#     count_step = 0
#     count_zero = 0
#     while s != "1":
#         count_zero += s.count("0")  # 0의 개수 업데이트
#         s = s.replace("0", "")  # 0 제거
#         count_step += 1
#         s = str(bin(len(s))[2:])
            
#     return [count_step, count_zero]
